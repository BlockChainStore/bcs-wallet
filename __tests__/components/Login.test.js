import React from 'react'
import configureStore from 'redux-mock-store'
import { Provider } from 'react-redux'
import { BrowserRouter } from 'react-router-dom'
import { shallow, mount } from 'enzyme'

import Login from '../../app/containers/LoginPrivateKey/LoginPrivateKey'

const setup = (shallowRender = true, state = { account: {
  loggedIn: true,
  wif: undefined
}}) => {
  const store = configureStore()(state)

  let wrapper
  if (shallowRender) {
    wrapper = shallow(<Login store={store} />)
  } else {
    wrapper = mount(
      <Provider store={store}>
        <BrowserRouter>
          <Login store={store} />
        </BrowserRouter>
      </Provider>
    )
  }

  return {
    store,
    wrapper
  }
}

describe('Login', () => {
  test('renders without crashing', () => {
    const { wrapper } = setup()
    expect(wrapper).toMatchSnapshot()
  })

  test('clicking show key icon toggles private key visibility', () => {
    const { wrapper } = setup(false)

    expect(wrapper.find('input').get(0).props.type).toEqual('password')

    wrapper
      .find('.viewKey')
      .first()
      .simulate('click')

    expect(wrapper.find('input').get(0).props.type).toEqual('text')
  })

  // test('private key field input onChange dispatches LOGIN action', () => {
  //   const { store, wrapper } = setup();
  //   let preventDefault = jest.fn();
  //   const deeperWrapper = wrapper.shallow();
  //
  //   const someWif = 'L1xpshXfzF6iQTq42onA5km8qwyzBaNQzPADhfTt2jzzcQSVoP5A'
  //   deeperWrapper
  //     .find('input')
  //     .simulate('change', { target: { value: someWif } })
  //     .find('button')
  //     .simulate('click');
  //
  //   expect(store.getActions()).toEqual([
  //     { wif: someWif, type: 'LOGIN' }
  //   ]);
  // });
})
