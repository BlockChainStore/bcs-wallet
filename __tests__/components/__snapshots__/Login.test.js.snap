// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login renders without crashing 1`] = `
<div
  className="loginPage"
  id="loginPage"
>
  <div
    className="title"
  >
    Login using a private key:
  </div>
  <form
    onSubmit={[Function]}
  >
    <div
      className="loginForm"
    >
      <PasswordField
        autoFocus={true}
        onChange={[Function]}
        placeholder="Enter your private key here (WIF)"
      />
    </div>
    <div>
      <Button
        cancel={false}
        disabled={true}
        primary={true}
        secondary={false}
        type="submit"
      >
        Login
      </Button>
      <HomeButtonLink />
    </div>
  </form>
</div>
`;
